datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -0,0 +1,159 @@
+                    query_filters.append(SearchAPISkipFilter(filter_input))
+                else:
+                    raise FilterError(
+                        "No valid filter name given within filter query param",
+                    )
+        elif query_param_name == "where":
+            # For the count endpoints
+            query_filters.extend(
+                SearchAPIQueryFilterFactory.get_query_filter(
+                    {"filter": request_filter}, entity_name,
+                ),
+            )
+        else:
+            raise FilterError(
+                f"Bad filter, please check query parameters: {request_filter}",
+            )
+
+        return query_filters
+
+    @staticmethod
+    def get_where_filter(filter_input, entity_name):
+        where_filters = []
+        if (
+            list(filter_input.keys())[0] == "and"
+            or list(filter_input.keys())[0] == "or"
+        ):
+            boolean_operator = list(filter_input.keys())[0]
+            conditions = list(filter_input.values())[0]
+            for condition in conditions:
+                # Could be nested AND/OR
+                where_filter = {
+                    "filter": {"where": condition},
+                }
+                conditional_where_filters = SearchAPIQueryFilterFactory.get_query_filter(
+                    where_filter, entity_name,
+                )
+
+                for conditional_where_filter in conditional_where_filters:
+                    conditional_where_filter.boolean_operator = boolean_operator
+                where_filters.extend(conditional_where_filters)
+        elif list(filter_input.keys())[0] == "text":
+            # TODO - we might want to move this to the data
+            # definitions at a later point
+            text_operator_fields = {
+                "datasets": ["title"],
+                "documents": ["title", "summary"],
+                "files": ["name"],
+                "instrument": ["name", "facility"],
+                "samples": ["name", "description"],
+                "techniques": ["name"],
+            }
+
+            try:
+                or_conditional_filters = []
+                field_names = text_operator_fields[entity_name]
+                for field_name in field_names:
+                    or_conditional_filters.append({field_name: filter_input["text"]})
+
+                where_filter = {
+                    "filter": {"where": {"or": or_conditional_filters}},
+                }
+                where_filters.extend(
+                    SearchAPIQueryFilterFactory.get_query_filter(
+                        where_filter, entity_name,
+                    ),
+                )
+            except KeyError:
+                # Do not raise FilterError nor attempt to create filters. Simply
+                # ignore text operator queries on fields that are not part of the
+                # text_operator_fields dict.
+                pass
```

Comment:
This is what the example Search API does.

Reply:
I'm slightly confused, are you agreeing with the comment or do we need to do something with this?

Reply:
This was meant to serve as a note in case you were wondering why we are not doing anything here. It is what the example Search API does so I have done the same. We don't need to do anything.